### PR TITLE
refactor(adapters-opencode-sdk): route and harden session todo loading via client

### PR DESCRIPTION
**Pull request by Maxsky5**
## Summary
- route `loadSessionTodos` through `ClientFactory` / SDK `session.todo` to match other message ops
- harden todo request failure handling with normalized warning payloads (`message`, `status`, `statusText`, `code` when available)
- preserve prior behavior by omitting `directory` when `workingDirectory` is blank
- refactor adapter tests to use explicit todo response modes (`success` / `api_error` / `throw`) and assert `createClient` inputs

## Validation
- bun run --filter @openducktor/adapters-opencode-sdk test
- bun run --filter @openducktor/adapters-opencode-sdk typecheck
- bun run --filter @openducktor/adapters-opencode-sdk lint

### CODE REVIEW

File: packages/adapters-opencode-sdk/src/message-ops.ts [hunk 1]
```diff
@@ -74,35 +131,36 @@ export const loadSessionHistory = async (
...
+      workingDirectory: input.workingDirectory,
     });
-    if (!response.ok) {
-      console.warn(`loadSessionTodos: HTTP ${response.status}`, {
-        statusText: response.statusText,
-      });
+    const response = await client.session.todo({
+      sessionID: input.externalSessionId,
+      ...(input.workingDirectory.trim().length > 0 ? { directory: input.workingDirectory } : {}),
+    });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/message-ops.ts:151]: Implemented in commit 36ee424. `loadSessionTodos` now computes `trimmedWorkingDirectory` once and forwards that trimmed value as `directory` when non-empty, so whitespace is never propagated to the SDK request.

File: packages/adapters-opencode-sdk/src/index.test.ts [hunk 2]
```diff
@@ -843,77 +898,141 @@ describe("OpencodeSdkAdapter", () => {
...
+    const mock = makeMockClient({
+      todoResult: {
+        mode: "success",
+        data: [],
+      },
+    });
+    const adapter = new OpencodeSdkAdapter({
+      createClient: () => mock.client,
+      now: () => "2026-02-17T12:00:00Z",
+    });
```
[the remark this answers is not in the document]
Comment (Maxsky5) [reply packages/adapters-opencode-sdk/src/index.test.ts:1026]: Implemented in commit 36ee424. The blank-`workingDirectory` test now also captures and asserts `createClient` arguments exactly as suggested. I also added an extra test that verifies whitespace-padded directories are trimmed before being sent to `session.todo`.

**Comment by Maxsky5** [comment]
Follow-up on non-inline feedback:

- Re: Gemini summary comment (https://github.com/Maxsky5/openducktor/pull/121#issuecomment-3980118953): informational summary only, no additional actionable defects beyond inline comments.
- Re: Greptile summary comment (https://github.com/Maxsky5/openducktor/pull/121#issuecomment-3980128247): informational/approval summary only, no code changes required.

Actionable inline items were implemented in commit `36ee424` and corresponding threads were resolved.
